apps/evm/src/ui/swap/cross-chain/cross-chain-swap-trade-review-route.tsx

```diff
@@ -0,0 +1,107 @@
+'use client'
+
+import { Currency, SkeletonBox, SkeletonText, SquidIcon } from '@sushiswap/ui'
+import { SushiXSwap2Adapter } from 'src/lib/swap/useCrossChainTrade/SushiXSwap2'
+import { Chain } from 'sushi/chain'
+import { STARGATE_TOKEN } from 'sushi/config'
+import { TradeRoutePathView } from '../trade-route-path-view'
+import {
+  useCrossChainSwapTrade,
+  useDerivedStateCrossChainSwap,
+} from './derivedstate-cross-chain-swap-provider'
+
+export const CrossChainSwapTradeReviewRoute = () => {
+  const {
+    state: { adapter, chainId0, token0, token1, chainId1 },
+  } = useDerivedStateCrossChainSwap()
+  const { data: trade, isLoading: isTradeLoading } = useCrossChainSwapTrade()
+
+  return (
+    <div className="px-3 flex justify-between w-full">
+      <div className="p-3 flex flex-col gap-2.5 overflow-hidden">
+        <span className="text-[10px] text-muted-foreground overflow-hidden overflow-ellipsis whitespace-nowrap">
+          From: {Chain.fromChainId(chainId0)?.name?.toUpperCase()}
+        </span>
+        <div className="flex flex-col gap-2">
+          {isTradeLoading || !trade ? (
```

Comment:
loading state on error (`isTradeLoading` false and `trade` null) doesn't make a lot of sense

also, maybe `isInitialLoading`? if, for some reason, the `useCrossChainSwapTrade` hook is disabled, it'll still do `isLoading` true